components/dashboard/src/whatsnew/WhatsNew.tsx

```diff
@@ -0,0 +1,100 @@
+]
+
+export const shouldSeeWhatsNew = (user: User, news: { newsKey: string, maxUserCreationDate: string }[] = allEntries) => {
+    const whatsNewSeen = user?.additionalData?.whatsNewSeen;
+    return news.some(x => user.creationDate <= x.maxUserCreationDate && (!whatsNewSeen || !whatsNewSeen[x.newsKey]));
+}
+
+export function WhatsNew(props: { onClose: () => void }) {
+    const { user, setUser } = useContext(UserContext);
+
+    const _unseenEntries = allEntries.filter(x => user && shouldSeeWhatsNew(user, [x])) || [];
+    const [visibleEntry, setVisibleEntry] = useState(_unseenEntries.pop());
+    const [unseenEntries, setUnseenEntries] = useState(_unseenEntries);
+
+    const markAsSeen = async (user?: User, ...news: (WhatsNewEntry | undefined)[]) => {
+        if (!user) {
+            return;
+        }
+
+        for (const n of news.filter(x => x && x.actionAfterSeen)) {
+            user = await n!.actionAfterSeen!(user);
+        };
+
+        const additionalData = user.additionalData = user.additionalData || {};
+        additionalData.whatsNewSeen = additionalData.whatsNewSeen || {};
+        const now = new Date().toISOString();
+        for (const newsKey of (news.filter(x => x !== undefined) as WhatsNewEntry[]).map(x => x.newsKey)) {
+            additionalData.whatsNewSeen[newsKey] = now;
+        }
+        user = await getGitpodService().server.updateLoggedInUser({
+            additionalData
+        });
+        setUser(user);
+    };
+
+    const internalClose = async () => {
+        await markAsSeen(user, ...unseenEntries, visibleEntry);
+        props.onClose();
+    };
+
+    const hasNext = () => unseenEntries.length > 0;
+
+    const next = async () => {
+        if (unseenEntries.length === 0) {
+            return;
+        }
+        visibleEntry && await markAsSeen(user, visibleEntry);
+        const _unseenEntries = unseenEntries;
+        setVisibleEntry(_unseenEntries.pop());
+        setUnseenEntries(_unseenEntries);
+    };
+
+    return <Modal
+        visible={!!visibleEntry}
+        onClose={internalClose}
+    >
+        <h3 className="pb-4">What's New 🎁</h3>
+        <>{visibleEntry && user ? visibleEntry.children(user, setUser) : <></>}</>
+        {
+            hasNext() ?
+                <>
+                    <div className="flex items-start justify-end mt-6 space-x-2">
+                        <button className="ml-2 mt-0 secondary" onClick={internalClose}>Dismiss All</button>
+                        <div className="flex flex-col items-center">
+                            <button className="ml-2" onClick={next}>Next</button>
+                            <div className="text-xs italic mt-1 ml-2 -mb-2">({unseenEntries.length} left)</div>
+                        </div>
+                    </div>
+                </>
+                :
+                <div className="flex justify-end mt-6 space-x-2">
+                    <button className="ml-2" onClick={internalClose}>Close</button>
```

Comment:
**question**: What do you think of renaming this to Continue? Left margin is also not needed here now, right?

```suggestion
                    <button onClick={internalClose}>Continue</button>
```

Alternatively, we could use the secondary class here with the _Close_ label.

```suggestion
                    <button className="secondary" onClick={internalClose}>Close</button>
```